test/test_cpp_standards.py

```diff
@@ -1,17 +1,89 @@
 import os
 
+import jinja2
 import pytest
 
-import utils
-
-project_dir = os.path.dirname(__file__)
-
-
-def test_cpp11():
+from . import utils
+from .template_projects import TemplateProject
+
+cpp_template_project = TemplateProject()
```

Comment:
This is so similar to the C template, that it'd be a pity if we couldn't merge it, no?

What's the difference here? Mainly `spam.cpp` vs.  `spam.c` (and `language="c++"`, but that's automatic, as mentioned above), I'd think?  `spam_cpp_top_level_add` is in the exact same spot as `spam_c_top_level_add`, I think, and `extra_compile_args` could be added to the base C template as well.

I mean ... it's not urgent/critical, because the current template approach is nicely extensible, but the C++ project was in this case really just some C++11/C++14/C++17 sprinkles on top of the default C project, so we could just as well show that in the code (if possible).

Reply:
Responded to this above, but I'm not all that happy with the template variables getting the wrong name in this scenario, and would quite like an example of a custom project in the code for reference.